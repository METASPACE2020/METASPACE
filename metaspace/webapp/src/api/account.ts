const fetchPostJson = async (url: string, body: object) =>
  await fetch(url, {
    method: 'POST',
    headers: {'Content-Type': 'application/json'},
    body: JSON.stringify(body),
  });

export const signInByEmail = async (email: string, password: string): Promise<boolean> => {
  const response = await fetchPostJson('/auth/signin', { email, password });
  return response.status >= 200 && response.status < 300;
};

export const createAccountByEmail = async (email: string, password: string, name: string) => {
  const response = await fetchPostJson('/auth/createaccount', { email, password, name });
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
};

export const sendPasswordResetToken = async (email: string) => {
  const response = await fetchPostJson('/auth/sendpasswordresettoken', { email });
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
};

export const validatePasswordResetToken = async (token: string, email: string): Promise<boolean> => {
  const response = await fetchPostJson('/auth/validatepasswordresettoken', { token, email });
  return response.status === 200;
};


export const resetPassword = async (token: string, email: string, password: string) => {
  const response = await fetchPostJson('/auth/sendpasswordreset', { token, email, password });
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
};
