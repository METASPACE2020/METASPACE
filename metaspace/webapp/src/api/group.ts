import gql from 'graphql-tag';

export type UserGroupRole = 'INVITED' | 'PENDING' | 'MEMBER' | 'PRINCIPAL_INVESTIGATOR';



export const createGroupMutation =
  gql`mutation createGroup($groupDetails: CreateGroupInput!) {
    createGroup(groupDetails: $groupDetails) {
      id
      name
      shortName
    }
  }`;


export const updateGroupMutation =
  gql`mutation updateGroup($id: ID!, $groupDetails: UpdateGroupInput!) {
    updateGroup(id: $id, groupDetails: $groupDetails) {
      id
      name
      shortName
    }
  }`;

export const deleteGroupMutation =
  gql`mutation deleteGroup($id: ID!) {
    deleteGroup(id: $id)
  }`;

export const removeUserFromGroupMutation =
  gql`mutation removeUserFromGroup($groupId: ID!, $userId: ID!) {
    removeUserFromGroup(groupId: $groupId, userId: $userId)
  }`;

export const requestAccessToGroupMutation =
  gql`mutation requestAccessToGroup($groupId: ID!, $bringDatasets: [ID!]!) { 
    requestAccessToGroup(groupId: $groupId, bringDatasets: $bringDatasets) {
      role # GraphQL demands a return value be specified
    }
  }`;

export const acceptRequestToJoinGroupMutation =
  gql`mutation acceptRequestToJoinGroup($groupId: ID!, $userId: ID!) {
    acceptRequestToJoinGroup(groupId: $groupId, userId: $userId) {
      role
    }
  }`;

export const inviteUserToGroupMutation =
  gql`mutation inviteUserToGroup($groupId: ID!, $email: String!) {
    inviteUserToGroup(groupId: $groupId, email: $email) {
      role
    }
  }`;

export const acceptGroupInvitationMutation =
  gql`mutation acceptGroupInvitation($groupId: ID!, $bringDatasets: [ID!]!) { 
    acceptGroupInvitation(groupId: $groupId, bringDatasets: $bringDatasets) {
      role
    }
  }`;

export const leaveGroupMutation =
  gql`mutation leaveGroup($groupId: ID!) { 
    leaveGroup(groupId: $groupId)
  }`;


export const editGroupQuery =
  gql`query EditGroupProfileQuery($groupId: ID!) {
    group(id: $groupId) {
      id
      name
      shortName
      members {
        role
        numDatasets
        user {
          id
          name
          email
        }
      }
    }
  }`;

export interface EditGroupQuery {
  id: string;
  name: string;
  shortName: string;
  members: EditGroupQueryMember[] | null;
}
export interface EditGroupQueryMember {
  role: UserGroupRole,
  numDatasets: number,
  user: EditGroupQueryUser
}
export interface EditGroupQueryUser {
  id: string;
  name: string;
  email: string | null;
}
