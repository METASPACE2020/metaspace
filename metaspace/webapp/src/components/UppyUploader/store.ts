// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
