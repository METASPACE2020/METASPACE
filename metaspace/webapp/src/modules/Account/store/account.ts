import { Module } from 'vuex';
import { DialogType } from '../dialogs';
import router from '../../../router';

export interface AccountState {
  dialog: DialogType | null;
  showDialogAsPage: boolean;
}

const account = {
  namespaced: true,
  state: {
    dialog: null,
    showDialogAsPage: false
  },
  mutations: {
    showDialog(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = false;
    },
    showDialogAsPage(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = true;
    },
    hideDialog(state: AccountState, dialog: DialogType) {
      if (state.dialog === dialog) {
        state.dialog = null;
        if (state.showDialogAsPage) {
          router.push('/');
        }
      }
    }
  }
};

export default account as Module<AccountState, any>;
