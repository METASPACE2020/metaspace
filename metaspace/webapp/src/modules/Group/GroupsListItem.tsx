import { defineComponent, computed } from 'vue'
import { useQuery } from '@vue/apollo-composable'
import { countGroupDatasets, UserGroupRoleOptions } from '../../api/group'
import { plural } from '../../lib/vueFilters'
import { useRouter } from 'vue-router'
import { encodeParams } from '../Filters'
import CopyButton from '../../components/CopyButton.vue'
import { ElIcon } from '../../lib/element-plus'
import { PictureFilled, EditPen } from '@element-plus/icons-vue'
import './GroupsListItem.scss'

interface GroupListItemProps {
  id: string
  name: string
  shortName: string
  currentUserRole: string
  urlSlug: string
  numMembers: number
}

export default defineComponent<GroupListItemProps>({
  name: 'GroupsListItem',
  props: {
    id: String,
    name: String,
    shortName: String,
    urlSlug: String,
    currentUserRole: String,
    numMembers: Number,
  },
  setup(props) {
    const router = useRouter()
    const queryVars = computed(() => ({
      groupId: props.id,
    }))

    const { result: countDatasetsResult } = useQuery(countGroupDatasets, queryVars)
    const datasetCount = computed(() => countDatasetsResult.value?.countDatasets)
    const groupLink = computed(() => ({
      name: 'group',
      params: { groupIdOrSlug: props.urlSlug || props.id },
    }))

    const datasetsLink = computed(() => ({
      path: '/datasets',
      query: encodeParams({ group: props.id }),
    }))

    const managementLink = computed(() => ({
      name: 'group',
      params: { groupIdOrSlug: props.urlSlug || props.id },
      query: { tab: props.currentUserRole === UserGroupRoleOptions.GROUP_ADMIN ? 'settings' : 'members' },
    }))

    const handleNavigation = async (params: any) => {
      await router.push(params)
    }

    return () => {
      const { id, name, shortName, numMembers, currentUserRole } = props
      const nOfDatasets: number = (datasetCount.value || 0) as unknown as number

      return (
        <div class="group-item">
          <div class="group-item-info">
            <div class="group-item-title-wrapper group-item-info-line">
              <div
                data-test-key="group-link"
                onClick={() => handleNavigation(groupLink.value)}
                class="group-item-title"
              >
                {name}
                {shortName ? <span class="group-item-short-name"> ({shortName})</span> : ''}
              </div>
              <CopyButton class="ml-1" isId text={id}>
                Copy group id to clipboard
              </CopyButton>
            </div>
            <div class="group-item-info-line">
              {nOfDatasets > 0 && (
                <div onClick={() => handleNavigation(datasetsLink.value)} class="group-item-title">
                  {plural(nOfDatasets, 'Dataset', 'Datasets')},
                </div>
              )}{' '}
              {plural(numMembers, 'Member', 'Members')}
            </div>
          </div>
          <div class="group-item-actions">
            <div class="flex items-center">
              <ElIcon>
                <PictureFilled />
              </ElIcon>
              <div class="link ml-1" data-test-key="dataset-link" onClick={() => handleNavigation(datasetsLink.value)}>
                Browse datasets
              </div>
            </div>
            <div class="flex items-center">
              <ElIcon>
                <EditPen />
              </ElIcon>
              <div class="link ml-1" data-test-key="manage-link" onClick={() => handleNavigation(managementLink.value)}>
                {currentUserRole === UserGroupRoleOptions.GROUP_ADMIN ? 'Manage' : 'Browse'} group
              </div>
            </div>
          </div>
        </div>
      )
    }
  },
})
