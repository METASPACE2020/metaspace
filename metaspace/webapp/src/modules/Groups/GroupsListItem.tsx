import { computed, defineComponent, reactive } from '@vue/composition-api'
import './GroupsListItem.scss'
import Vue from 'vue'
import { encodeParams } from '../Filters'
import CopyButton from '../../components/CopyButton.vue'
import { useQuery } from '@vue/apollo-composable'
import { countGroupDatasets, getUserGroupsQuery, ViewGroupResult } from '../../api/group'
import { plural } from '../../lib/vueFilters'

const RouterLink = Vue.component('router-link')

interface GroupListItemProps {
  id: string
  name: string
  shortName: string
  urlSlug: string
  numMembers: number
}

interface GroupListItemState {
  groupNameFilter: string | undefined
}

export default defineComponent<GroupListItemProps>({
  name: 'GroupsListItem',
  props: {
    id: {
      type: String,
    },
    name: {
      type: String,
    },
    shortName: {
      type: String,
    },
    urlSlug: {
      type: String,
    },
    numMembers: {
      type: Number,
    },
  },
  setup: function(props, ctx) {
    const { $route, $store } = ctx.root
    const state = reactive<GroupListItemState>({
      groupNameFilter: '',
    })

    const queryVars = computed(() => ({
      groupId: props.id,
    }))
    const {
      result: countDatasetsResult,
    } = useQuery<Number|any>(countGroupDatasets, queryVars)
    const datasetCount = computed(() => countDatasetsResult.value != null ? countDatasetsResult.value.countDatasets
      : null)

    const groupLink = (id: String, urlSlug: String) => {
      return {
        name: 'group',
        params: { groupIdOrSlug: urlSlug || id },
      }
    }

    const datasetsLink = (id: String) => {
      return {
        path: '/datasets',
        query: encodeParams({ group: id }),
      }
    }

    const managementLink = (id: String, urlSlug: String) => {
      return {
        name: 'group',
        params: { groupIdOrSlug: urlSlug || id },
        query: { tab: 'settings' },
      }
    }

    return () => {
      const { id, name, shortName, urlSlug, numMembers } = props
      const nOfDatasets : number = (datasetCount.value || 0) as unknown as number

      return (
        <div class='group-item'>
          <div class="group-item-info">
            <div class="group-item-title-wrapper group-item-info-line">
              <RouterLink to={groupLink(id, urlSlug)} class='group-item-title'>
                {name}{shortName ? <span class='group-item-short-name'> ({shortName})</span> : ''}
              </RouterLink>
              <CopyButton
                class="ml-1"
                isId
                text={id}>
                Copy group id to clipboard
              </CopyButton>
            </div>
            <div class="group-item-info-line">
              {
                nOfDatasets > 0
                && <RouterLink to={datasetsLink(id)} class='group-item-title'>
                  {plural(nOfDatasets, 'Dataset', 'Datasets')},
                </RouterLink>
              }
              {plural(numMembers, 'Member', 'Members') }
            </div>
          </div>
          <div class="group-item-actions">
            <div>
              <i class="el-icon-picture"/>
              <RouterLink to={datasetsLink(id)} class='ml-1'>
                Browse datasets
              </RouterLink>
            </div>
            <div>
              <i class="el-icon-edit" />
              <RouterLink to={managementLink(id, urlSlug)} class='ml-1'>
                Manage group
              </RouterLink>
            </div>
          </div>
        </div>
      )
    }
  },
})
