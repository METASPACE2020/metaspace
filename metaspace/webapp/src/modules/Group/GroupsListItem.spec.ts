import { defineComponent, nextTick, h, ref } from 'vue'
import { flushPromises, mount } from '@vue/test-utils'
import GroupsListItem from './GroupsListItem'
import { initMockGraphqlClient } from '../../tests/utils/mockGraphqlClient'
import { UserGroupRoleOptions } from '../../api/group'
import { vi } from 'vitest'
import { DefaultApolloClient, useQuery } from '@vue/apollo-composable'
import store from '../../store'
import router from '../../router'

vi.mock('@vue/apollo-composable', () => ({
  useQuery: vi.fn(),
  useSubscription: vi.fn(() => ({ onResult: vi.fn() })),
  DefaultApolloClient: vi.fn(),
}))

let graphqlMocks: any
let mockRoutePush

describe('GroupsListItem', () => {
  const propsData = {
    id: 'gr1',
    name: 'Group 1',
    shortName: 'grp1',
    urlSlug: 'grp1',
    numMembers: 1,
    currentUserRole: UserGroupRoleOptions.GROUP_ADMIN,
  }

  const testHarness = defineComponent({
    components: {
      GroupsListItem,
    },
    props: ['id', 'name', 'shortName', 'urlSlug', 'currentUserRole', 'numMembers'],
    setup(props) {
      return () => h(GroupsListItem, { ...props })
    },
  })

  const graphqlWithData = async () => {
    const params = {
      countDatasets: () => 1,
    }

    graphqlMocks = await initMockGraphqlClient({
      Query: () => params,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(params).reduce((acc, key) => ({ ...acc, [key]: params[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
    })
  }

  const graphqlWithNoData = async () => {
    const params = {
      countDatasets: () => 0,
    }

    graphqlMocks = await initMockGraphqlClient({
      Query: () => params,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(params).reduce((acc, key) => ({ ...acc, [key]: params[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
    })
  }

  beforeAll(async () => {
    await graphqlWithData()
  })

  beforeEach(() => {
    mockRoutePush = vi.fn()
    vi.mock('vue-router', async () => {
      const actual: any = await vi.importActual('vue-router')
      return {
        ...actual,
        useRouter: () => {
          return {
            push: mockRoutePush,
          }
        },
      }
    })
  })

  it('it should match snapshot with one member and admin role', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: propsData,
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it should match snapshot with two members and admin role', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: { ...propsData, numMembers: 2 },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it should match snapshot with two members and not admin role', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: {
        ...propsData,
        currentUserRole: UserGroupRoleOptions.MEMBER,
      },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it should match snapshot with two members, not admin role and 0 datasets', async () => {
    await graphqlWithNoData()
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: {
        ...propsData,
        currentUserRole: UserGroupRoleOptions.MEMBER,
      },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render the correct navigation links if admin', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: propsData,
    })
    await flushPromises()
    await nextTick()

    wrapper.find('[data-test-key="group-link"]').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      name: 'group',
      params: { groupIdOrSlug: propsData.urlSlug },
    })

    wrapper.find('[data-test-key="manage-link"]').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      name: 'group',
      params: { groupIdOrSlug: propsData.urlSlug },
      query: { tab: 'settings' },
    })

    wrapper.find('[data-test-key="dataset-link"]').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      path: '/datasets',
      query: { grp: propsData.id },
    })
  })

  it('should render the correct navigation links if not admin', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: {
        ...propsData,
        currentUserRole: UserGroupRoleOptions.MEMBER,
      },
    })
    await flushPromises()
    await nextTick()

    wrapper.find('[data-test-key="group-link"]').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      name: 'group',
      params: { groupIdOrSlug: propsData.urlSlug },
    })

    wrapper.find('[data-test-key="manage-link"]').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      name: 'group',
      params: { groupIdOrSlug: propsData.urlSlug },
      query: { tab: 'members' },
    })

    wrapper.find('[data-test-key="dataset-link"]').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      path: '/datasets',
      query: { grp: propsData.id },
    })
  })
})
