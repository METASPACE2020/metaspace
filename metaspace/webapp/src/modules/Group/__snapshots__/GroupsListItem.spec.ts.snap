// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GroupsListItem > it should match snapshot with one member and admin role 1`] = `
"<div class=\\"group-item\\">
  <div class=\\"group-item-info\\">
    <div class=\\"group-item-title-wrapper group-item-info-line\\">
      <div data-test-key=\\"group-link\\" class=\\"group-item-title\\">Group 1<span class=\\"group-item-short-name\\"> (grp1)</span></div>
      <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover ml-1\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
        <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy group id to clipboard</p>
      </div>
    </div>
    <div class=\\"group-item-info-line\\">
      <div class=\\"group-item-title\\">1 Dataset,</div> 1 Member
    </div>
  </div>
  <div class=\\"group-item-actions\\">
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"M96 896a32 32 0 0 1-32-32V160a32 32 0 0 1 32-32h832a32 32 0 0 1 32 32v704a32 32 0 0 1-32 32H96zm315.52-228.48-68.928-68.928a32 32 0 0 0-45.248 0L128 768.064h778.688l-242.112-290.56a32 32 0 0 0-49.216 0L458.752 665.408a32 32 0 0 1-47.232 2.112zM256 384a96 96 0 1 0 192.064-.064A96 96 0 0 0 256 384z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"dataset-link\\">Browse datasets</div>
    </div>
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"manage-link\\">Manage group</div>
    </div>
  </div>
</div>"
`;

exports[`GroupsListItem > it should match snapshot with two members and admin role 1`] = `
"<div class=\\"group-item\\">
  <div class=\\"group-item-info\\">
    <div class=\\"group-item-title-wrapper group-item-info-line\\">
      <div data-test-key=\\"group-link\\" class=\\"group-item-title\\">Group 1<span class=\\"group-item-short-name\\"> (grp1)</span></div>
      <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover ml-1\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
        <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy group id to clipboard</p>
      </div>
    </div>
    <div class=\\"group-item-info-line\\">
      <div class=\\"group-item-title\\">1 Dataset,</div> 2 Members
    </div>
  </div>
  <div class=\\"group-item-actions\\">
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"M96 896a32 32 0 0 1-32-32V160a32 32 0 0 1 32-32h832a32 32 0 0 1 32 32v704a32 32 0 0 1-32 32H96zm315.52-228.48-68.928-68.928a32 32 0 0 0-45.248 0L128 768.064h778.688l-242.112-290.56a32 32 0 0 0-49.216 0L458.752 665.408a32 32 0 0 1-47.232 2.112zM256 384a96 96 0 1 0 192.064-.064A96 96 0 0 0 256 384z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"dataset-link\\">Browse datasets</div>
    </div>
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"manage-link\\">Manage group</div>
    </div>
  </div>
</div>"
`;

exports[`GroupsListItem > it should match snapshot with two members and not admin role 1`] = `
"<div class=\\"group-item\\">
  <div class=\\"group-item-info\\">
    <div class=\\"group-item-title-wrapper group-item-info-line\\">
      <div data-test-key=\\"group-link\\" class=\\"group-item-title\\">Group 1<span class=\\"group-item-short-name\\"> (grp1)</span></div>
      <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover ml-1\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
        <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy group id to clipboard</p>
      </div>
    </div>
    <div class=\\"group-item-info-line\\">
      <div class=\\"group-item-title\\">1 Dataset,</div> 1 Member
    </div>
  </div>
  <div class=\\"group-item-actions\\">
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"M96 896a32 32 0 0 1-32-32V160a32 32 0 0 1 32-32h832a32 32 0 0 1 32 32v704a32 32 0 0 1-32 32H96zm315.52-228.48-68.928-68.928a32 32 0 0 0-45.248 0L128 768.064h778.688l-242.112-290.56a32 32 0 0 0-49.216 0L458.752 665.408a32 32 0 0 1-47.232 2.112zM256 384a96 96 0 1 0 192.064-.064A96 96 0 0 0 256 384z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"dataset-link\\">Browse datasets</div>
    </div>
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"manage-link\\">Browse group</div>
    </div>
  </div>
</div>"
`;

exports[`GroupsListItem > it should match snapshot with two members, not admin role and 0 datasets 1`] = `
"<div class=\\"group-item\\">
  <div class=\\"group-item-info\\">
    <div class=\\"group-item-title-wrapper group-item-info-line\\">
      <div data-test-key=\\"group-link\\" class=\\"group-item-title\\">Group 1<span class=\\"group-item-short-name\\"> (grp1)</span></div>
      <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover ml-1\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
        <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy group id to clipboard</p>
      </div>
    </div>
    <div class=\\"group-item-info-line\\">
      <!----> 1 Member
    </div>
  </div>
  <div class=\\"group-item-actions\\">
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"M96 896a32 32 0 0 1-32-32V160a32 32 0 0 1 32-32h832a32 32 0 0 1 32 32v704a32 32 0 0 1-32 32H96zm315.52-228.48-68.928-68.928a32 32 0 0 0-45.248 0L128 768.064h778.688l-242.112-290.56a32 32 0 0 0-49.216 0L458.752 665.408a32 32 0 0 1-47.232 2.112zM256 384a96 96 0 1 0 192.064-.064A96 96 0 0 0 256 384z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"dataset-link\\">Browse datasets</div>
    </div>
    <div class=\\"flex items-center\\">
      <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
          <path fill=\\"currentColor\\" d=\\"m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z\\"></path>
        </svg></div>
      <div class=\\"link ml-1\\" data-test-key=\\"manage-link\\">Browse group</div>
    </div>
  </div>
</div>"
`;
